formats/properties/commonMain/src/kotlinx/serialization/properties/Properties.kt

```diff
@@ -118,7 +165,18 @@ public sealed class Properties(
      */
     @ExperimentalSerializationApi
     public fun <T> decodeFromMap(deserializer: DeserializationStrategy<T>, map: Map<String, Any>): T {
-        val m = InMapper(map, deserializer.descriptor)
+        val m = InAnyMapper(map, deserializer.descriptor)
+        return m.decodeSerializableValue(deserializer)
+    }
+
+    /**
+     * Decodes properties from the given [map] to a value of type [T] using the given [deserializer].
+     * All values are expected to be of type [String].
```

Comment:
`All values are expected to be of type [String]` — this phrase looks redundant, because this fact is already expressed by type `Map<String, String>`. 

I'd replace it with the phrase indicating that strings are then converted to target kotlin type using default conversions.

Reply:
Applied suggestion.